woodwork/table_accessor.py

```diff
@@ -164,6 +164,37 @@ def set_index(self, new_index):
             _check_index(self._dataframe, self._schema.index)
         self._set_underlying_index()
 
+    def set_types(self, logical_types=None, semantic_tags=None, retain_index_tags=True):
+        """Update the logical tpye and semantic tags for any columns names in the provided types dictionaries,
+        updating the Woodwork typing information for the DataFrame.
+
+        Args:
+            logical_types (dict[str -> str], optional): A dictionary defining the new logical types for the
+                specified columns.
+            semantic_tags (dict[str -> str/list/set], optional): A dictionary defining the new semantic_tags for the
+                specified columns.
+            retain_index_tags (bool, optional): If True, will retain any index or time_index
+                semantic tags set on the column. If False, will replace all semantic tags any time a column's
+                semantic tags or logical type changes. Defaults to True.
+        """
+        logical_types = logical_types or {}
+        logical_types = {col_name: _get_column_logical_type(None, ltype, col_name) for col_name, ltype in logical_types.items()}
```

Comment:
Felt slightly weird using `_get_column_logical_type` here, since, technically, if `ltype` wasn't defined it'd try to infer the logical type, which we definitely don't want (this is happening in the DataTables' set types as well). To avoid that, I passed `None` in for the series so that we definitely wouldn't quietly infer a LogicalType during `set_types`.

I'd be open to pulling out the logic from this function that parses a logical type user input, but not sure how useful that'd be.

Reply:
My initial reaction is that we should pull out the logic that parses the logical type into a separate function that we could use here and inside `_get_column_logical_type`. That would make the code a bit more understandable I think rather than trying to understand why `None` is being passed for the series here.

Reply:
Cool. I'm happy to pull that logic out! (and also maybe just move it to the Schema.set_types)